include/TrustWalletCore/TWHDWallet.h

```diff
@@ -25,17 +25,17 @@ struct TWHDWallet;
 TW_EXPORT_STATIC_METHOD
 bool TWHDWalletIsValid(TWString *_Nonnull mnemonic);
 
-/// Creates a new random HDWallet with the provided strength in bits.  Returned object needs to be deleted.
+/// Creates a new random HDWallet with the provided strength in bits.  Null is returned on invalid strength.  Returned object needs to be deleted.
 TW_EXPORT_STATIC_METHOD
-struct TWHDWallet *_Nonnull TWHDWalletCreate(int strength, TWString *_Nonnull passphrase);
+struct TWHDWallet *_Nullable TWHDWalletCreate(int strength, TWString *_Nonnull passphrase);
 
-/// Creates an HDWallet from a mnemonic seed.  Returned object needs to be deleted.
+/// Creates an HDWallet from a mnemonic seed.  Null is returned on invalid mnemonic.  Returned object needs to be deleted.
 TW_EXPORT_STATIC_METHOD
 struct TWHDWallet *_Nullable TWHDWalletCreateWithMnemonic(TWString *_Nonnull mnemonic, TWString *_Nonnull passphrase);
 
-/// Creates an HDWallet from a seed.  Returned object needs to be deleted.
+/// Creates an HDWallet from a seed.  Null is returned on invalid input.  Returned object needs to be deleted.
 TW_EXPORT_STATIC_METHOD
-struct TWHDWallet *_Nonnull TWHDWalletCreateWithData(TWData *_Nonnull data, TWString *_Nonnull passphrase);
+struct TWHDWallet *_Nullable TWHDWalletCreateWithData(TWData *_Nonnull data, TWString *_Nonnull passphrase);
```

Comment:
Test this method "TWHDWalletCreateWithData" with android sample HDWallet(seed,""), will crash.

Reply:
I looked into this, and the reason is that when initialized with a binary data, that should be the `entropy`, which is the 1-to-1 binary representation of the mnemonic phrase.  The seed is different, it is derived from the mnemonic, it is longer, and reverse computation is not possible.  So instead of seed it should use the entropy field.  I just realize this field is not exposed through the C-Kotlin interface...
BTW, I think HDWallet will not technically crash, just return a null result.  Access that will crash.
I also realize the descriptions are inconsistent, as 'seed' is mentioned.  I will improve this, thanks for the note.